res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -275,32 +275,29 @@ TraktorS2MK3.syncHandler = function (field) {
         // Light LED while pressed
         TraktorS2MK3.outputHandler(field.value, field.group, "sync_enabled");
     } else {
-        script.toggleControl(field.group, "beatsync");
-
         if (field.value) {
-            // Start timer to measure how long button is pressed
-            TraktorS2MK3.syncPressedTimer[field.group] = engine.beginTimer(300, function () {
-                // Reset sync button timer state if active
-                if (TraktorS2MK3.syncPressedTimer[field.group] !== 0) {
-                    TraktorS2MK3.syncPressedTimer[field.group] = 0;
-                }
-            }, true);
-
-            // Light corresponding LED when button is pressed
-            TraktorS2MK3.outputHandler(1, field.group, "sync_enabled");
-        } else {
-            if (engine.getValue(field.group, "sync_enabled")) {
+            if (engine.getValue(field.group, 'sync_enabled') === 0) {
+                script.triggerControl(field.group, "beatsync");
+                // Start timer to measure how long button is pressed
+                TraktorS2MK3.syncPressedTimer[field.group] = engine.beginTimer(300, function () {
+                    engine.setValue(field.group, "sync_enabled", 1);
+                    // Reset sync button timer state if active
+                    if (TraktorS2MK3.syncPressedTimer[field.group] !== 0) {
+                        TraktorS2MK3.syncPressedTimer[field.group] = 0;
+                    }
+                }, true);
+
+                // Light corresponding LED when button is pressed
+                TraktorS2MK3.outputHandler(1, field.group, "sync_enabled");
```

Comment:
This should go inside the timer callback when `sync_enabled` is set to 1.

Reply:
In general i light all LEDs on the controller in case a button is pressed, so the user has some visual feedback if the button was pressed. Also here, the LED is turned off, if the button is released again, except the timer is not running anymore and sync lock is activated.

Reply:
okay